v3/src/data-interactive/handlers/di-handler-utils.ts

```diff
@@ -17,6 +17,12 @@ export function createAttribute(value: DIAttribute, dataContext: IDataSet, colle
     const attribute = dataContext.addAttribute(attributeSnapshot, { collection: collection?.id })
     if (value.formula) attribute.formula?.setDisplayExpression(value.formula)
     metadata?.setIsHidden(attribute.id, !!value.hidden)
+    if (value.colormap != null && metadata) {
+        const categorySet = metadata.getCategorySet(attribute.id)
+        Object.entries(value.colormap as DICategoryColorMap || {}).forEach(([category, color]) => {
+          categorySet?.setColorForCategory(category, color ?? (categorySet.colorMap[category] || ""))
```

Comment:
If `DICategoryColorMap` didn't have `undefined` as a possible value type, this could just be:
```suggestion
          categorySet?.setColorForCategory(category, color)
```